Library/Homebrew/brew.rb

```diff
@@ -132,7 +132,9 @@ class MissingEnvironmentVariables < RuntimeError; end
     possible_tap = OFFICIAL_CMD_TAPS.find { |_, cmds| cmds.include?(cmd) }
     possible_tap = Tap.fetch(possible_tap.first) if possible_tap
 
-    odie "Unknown command: #{cmd}" if !possible_tap || possible_tap.installed?
+    if !possible_tap || possible_tap.installed? || Tap.untapped_official_taps.include?(possible_tap.name)
```

Comment:
So this logic (and everything related) means that if you untap it: it'll never be automatically retapped?

Reply:
Yes, that's the idea. Although, if you retap using `brew tap homebrew/cask`, it removes homebrew/cask from the `homebrew.untapped` setting meaning that it would be retapped automatically in the future if needed (assuming the reason it became untapped was not done using `tap.uninstall`. tbh I'm not sure how that would happen).

Actually, now that I'm thinking about it, are there any instances where a tap could be uninstalled and we _wouldn't_ want to write to the setting? Maybe we should only write if a user untaps via the `brew untap` command...

Reply:
> Yes, that's the idea. Although, if you retap using `brew tap homebrew/cask`, it removes homebrew/cask from the `homebrew.untapped` setting meaning that it would be retapped automatically in the future if needed (assuming the reason it became untapped was not done using `tap.uninstall`. tbh I'm not sure how that would happen).

Seems good!

> Maybe we should only write if a user untaps via the `brew untap` command...

Yeh, maybe?

Reply:
This line, in particular, is saying "raise an error saying unknown command if there is no tap found with the given command, the tap with the given command is already installed, or the tap with the given command has been manually uninstalled by the user. Otherwise, continue and autotap"